snuba/web/converters.py

```diff
@@ -0,0 +1,12 @@
+from werkzeug.routing import BaseConverter
+
+from snuba.datasets.dataset import Dataset
+from snuba.datasets.factory import get_dataset
+
+
+class DatasetConverter(BaseConverter):
+    def to_python(self, value: str):
+        return get_dataset(value)
```

Comment:
If `InvalidDatasetError` is raised here, it will be automatically translated to a 404: https://github.com/getsentry/snuba/blob/8e3e55def885ff13b53057e8ebe3dce8ec80fdec/snuba/web/views.py#L137-L144